Documentation/internals/security-identities.rst

```diff
@@ -0,0 +1,48 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+.. _security_identities:
+
+*******************
+Security Identities
+*******************
+
+Security identities are generated from labels. They are stored as ``uint16``,
+which means the maximum limit for a security identity is ``2^16 - 1``. The
+minimum security identity is ``256``, due to the reserved identity range being
+from ``1`` to ``255``.
+
+There are two caveats to the above however:
+
+1) ClusterMesh
+2) Identities generated from CIDR-based policies
+
+For (1), an extra 8 bits are added as the ``cluster-id`` which identifies the
+cluster in the ClusterMesh, into the 3rd octet as shown by ``0x00FF0000``. This
+makes the security identity's upper limit ``2^24 - 1``. This does not apply to
+CIDR identities however, see (2).
+
+For (2), CIDR identities are local to each node. This means that they're not
+subject to the global limit of security identities (``2^16 - 1``). CIDR
+identities begin from ``1`` and end at ``16777215``, however since they're
+shifted by ``24``, this makes their effective range ``1 | (1 << 24)`` to
+``16777215 | (1 << 24)``.
+
+.. note::
+
+   When CIDR policies are applied, the identity generated is local to each
+   node. In other words, the identity may not be the same for the same CIDR
+   policy across two nodes. Additionally, CIDR identities are not included
+   within the numerical identities range (``256 - 2^16 -1``), as it is stored
+   separately inside the Agent.
+
+Overall, the following represents the different ranges:
+
+.. code::
+
+   0x00000000 - 0x0000FFFF (0    to 2^16 - 1)        => cluster-local identities
```

Comment:
I think it's worth calling out that `0` is never explicitly set as a valid identity value, but rather if this shows up somewhere then this typically either means "we don't know the identity", or in a few cases "any identity" (for example, in policy in the eBPF datapath we set identity=0 with a port=nonzero to apply l4-only policy which applies to all identities.) . It doesn't have a fixed corresponding set of labels like any other identity.

Reply:
Good idea. I've added a section for this.